File: tests/verify/managed/fixtures/poisoned/poison.dart
// poison.dart — poisoned Dart fixture. Every commented rule id MUST fire.

@weftHot
void onSignal(double x, Float64List out) {
  final scratch = List<double>.filled(64, 0.0); // expect: WV-DT-002
  final box = Map<String, double>();            // expect: WV-DT-003
  final p = calloc<Float>(64);                  // expect: WV-DT-006
  final sb = StringBuffer();                    // expect: WV-DT-005
  final cb = () => out[0] + x;                  // expect: WV-DT-008
  final row = [x, x];                           // expect: WV-DT-004
  out[0] = scratch[0] + box['a']! + p[0] + sb.length + cb() + row[1];
}

File: tests/verify/managed/fixtures/poisoned/poison.rs
// poison.rs — poisoned Rust fixture. Every commented rule id MUST fire.

#[weft_hot]
fn on_quote(q: &Quote, out: &mut Out) {
    let boxed = Box::new(q.px);              // expect: WV-RS-001
    let v: Vec<u8> = vec![0u8; 64];          // expect: WV-RS-003
    let s = format!("{:?}", q.px);           // expect: WV-RS-009
    let owned = q.symbol.to_string();        // expect: WV-RS-006
    let cloned = out.scratch.clone();        // expect: WV-RS-008
    let acc: Vec<f64> = it.collect();        // expect: WV-RS-010
    let _ = (boxed, v, s, owned, cloned, acc);
}
